R/transformer.R

```diff
@@ -8,5 +8,28 @@
 #'   custom glue transformers and some common use cases.
 #' @export
 identity_transformer <- function(text, envir) {
-  eval(parse(text = text, keep.source = FALSE), envir)
+  if (requireNamespace("rlang", quiet = TRUE)) {
+    tryCatch(
+      expr <- parse(text = text, keep.source = FALSE),
+      error = function(err) {
+        rlang::abort(
+          "Failed to parse glue component",
+          parent = err,
+          call = NULL
+        )
+      }
+    )
+    tryCatch(
+      eval(expr, envir),
+      error = function(err) {
+        rlang::abort(
+          paste0("Failed to evaluate glue component {", text, "}"),
+          parent = err,
+          call = NULL
+        )
+      }
+    )
+  } else {
+    eval(parse(text = text, keep.source = FALSE), envir)
```

Comment:
How about this?

```r
identity_transformer <- function(text, envir) {
  expr <- with_rethrown_errors(
    parse(text = text, keep.source = FALSE),
    "Failed to parse glue component."
  )
  with_rethrown_errors(
    eval(expr, envir),
    paste0("Failed to evaluate glue component {", text, "}.")
  )
}

with_rethrown_errors <- function(expr, message) {
  if (!requireNamespace("rlang", quietly = TRUE)) {
    return(expr)
  }

  rlang::try_fetch(
    expr,
    error = function(cnd) {
      rlang::abort(
        message,
        parent = cnd,
        call = NULL
      )
    }
  )
}
```

Reply:
I happened to have this installed locally and am seeing `with_rethrown_errors()` come up quite a bit in profiling due to the usage of `try_fetch()`, so I'd suggest switching to `withCallingHandlers()` since it is in C and we just rethrow the error